lpcompiler.pas

```diff
@@ -1279,7 +1287,7 @@ function TLapeCompiler.ParseMethodHeader(out Name: lpString; addToScope: Boolean
       else if (LapeCase(Name) = LapeCase(AVar)) or hasDeclaration(AVar, True) then
         LapeExceptionFmt(lpeDuplicateDeclaration, [AVar], Tokenizer.DocPos)
       else
-        FStackInfo.addVar(ParType, VarType, AVar);
+        FStackInfo.addVar(ParType, VarType, AVar, not (lcoHints in FOptions));
```

Comment:
Not sure if it's worth adding a parameter for `StackInfo.addVar` for this. I think this is the only place where you need to set `.Used` to `False` / `Ignore`. You could probably get rid of the extra parameter, set `.Used` to `Ignore` by default, then set it to `False` here.